### DISCUSSION TITLE
Squeeze making wonders? Bias is -ve?

### DISCUSSION THREAD

**Discussion by MuhammadBilal848** [mrdbourke/pytorch-deep-learning#933] [Q&A]
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/57318407/c074f494-9931-4949-abe8-bd5314a148ce)

I wrote the Linear Regression Class a bit differently using `nn.Linear` method. The thing I noticed is that, in the video (Pytorch workflow), we were assigning the `nn.Parameters(torch.rand())` so rand wasn't assigning the `w` and `b` as negative, so the model was taking off from positive `w` and `b` and was able to successfully converge. 

But in `nn.Linear()`, the method assigns the `w` and `b` as negative, and from that point, the model makes efforts to converge and takes multiple attempts to finally make good predictions. I don't get it why?

Also, I tried removing the `.squeeze(1)` from the `forward()` method and it was also affecting the model's accuracy. could anyone explain this?
